src/app/components_ngrx/work-item-detail/work-item-detail.component.ts

```diff
@@ -78,6 +78,8 @@ export class WorkItemDetailComponent implements OnInit, OnDestroy, AfterViewChec
       this.listenToEsc = true;
     }
   }
+  @Input() workItemTypes: WorkItemTypeUI[] = [];
```

Comment:
Why taking input for work item types? We have `workItemTypeSource` that gives us an observable of the list of work item types. Besides, this component will also be used for the non-contextual detail page, there, where are you gonna get the input from?

Reply:
Or, you can use `this.wiTypes` instead.

Reply:
I didnt use `workItemTypeSource` or `this.wiTypes` because they return all the work item types and not work item types for particular type group and since the code for getting work item types for a particular type group has been already implemented in the planner list component so instead of adding it in work item detail component I thought of passing it as input to work item detail component. 

But since we have to add it in the non-contextual detail page I will add the code in work item detail component as well